src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -358,13 +356,6 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 	 */
 	if (viaDeprecatedAPI)
 	{
-		/*
-		 * We exit early but there is no need to close colocatedRelation. Because
-		 * if viaDeprecatedAPI is true, we never open colocatedRelation in the first
-		 * place.
-		 */
-		Assert(colocatedRelation == NULL);
```

Comment:
For this line, instead of removing we should maybe change the assert `colocateWithTableName == NULL` here?

Reply:
done